backend/src/core/middlewares/protected.middleware.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response, NextFunction } from 'express'
+import { ProtectedService } from '@core/services'
+import logger from '@core/logger'
+
+/**
+ * Ensure that the template body only has the necessary keywords if it is a protected campaign.
+ * Subject should not have any keywords.
+ * @param req
+ * @param res
+ * @param next
+ */
+const verifyTemplate = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<Response | void> => {
+  try {
+    // If it is not a protected campaign, move on to the next middleware
+    if (!res.locals.isProtected) {
```

Comment:
This is the only line that uses `res.locals.isProtected`, so you can call `ProtectedService.isProtectedCampaign(+req.params.campaignId)` here